web3/middleware/validation.py

```diff
@@ -27,20 +41,68 @@ def validate_chain_id(web3, chain_id):
         )
 
 
+def check_extradata_length(val):
+    if not isinstance(val, (str, int, bytes)):
+        return val
+    result = HexBytes(val)
+    if len(result) > MAX_EXTRADATA_LENGTH:
+        raise ValidationError(
+            "The field extraData is %d bytes, but should be %d. "
+            "It is quite likely that you are connected to a POA chain. "
+            "Refer "
+            "http://web3py.readthedocs.io/en/latest/middleware.html#geth-style-proof-of-authority "
+            "for more details. The full extraData is: %r" % (
+                len(result), MAX_EXTRADATA_LENGTH, result
+            )
+        )
+    return val
+
+
 def transaction_normalizer(transaction):
     return dissoc(transaction, 'chainId')
 
 
-def validation_middleware(make_request, web3):
-    transaction_validator = apply_formatters_to_dict({
-    })
+def transaction_param_validator(web3):
+    transactions_params_validators = {
+        'chainId': validate_chain_id(web3),
+    }
+    return apply_formatter_at_index(
+        apply_formatters_to_dict(transactions_params_validators),
+        0
+    )
+
+
+BLOCK_VALIDATORS = {
+    'extraData': check_extradata_length,
+}
+
+
+block_validator = apply_formatters_to_dict(BLOCK_VALIDATORS)
+
+
+@curry
+def chain_id_validator(web3):
+    return compose(
+        apply_formatter_at_index(transaction_normalizer, 0),
+        transaction_param_validator(web3)
+    )
+
+
+extra_data_validator = apply_formatter_if(is_not_null, block_validator)
+
+
+def build_validators_with_web3(w3):
+    return dict(
+        request_formatters={
+            'eth_sendTransaction': chain_id_validator(w3),
```

Comment:
After having a look at https://github.com/ethereum/web3.py/pull/777/files#diff-1e8b0fe1b6f6e15899cb2f41637945bbL36, I'm not sure if I should remove these 3 lines below or If I should comment out [line 67](https://github.com/ethereum/web3.py/pull/756/files#diff-1e8b0fe1b6f6e15899cb2f41637945bbR67).
If I  remove the lines below, it won't run the entire `transaction_param_validator` chain, However It won't  leave an example behind on how to use the validation middleware with w3 instance.

Reply:
One approach would be to leave the validator in, but short-circuit it if `w3.net.chainId is None`. In that case, we assume `chainId` is unknowable, so we just ignore its value in the transaction, and strip it out.

Reply:
done